psi4/driver/procrouting/proc.py

```diff
@@ -988,6 +989,46 @@ def select_mp4(name, **kwargs):
         return func(name, **kwargs)
 
 
+def select_adc2(name, **kwargs):
+    """Function selecting the algorithm for ADC(2) excited state energy
+    call and directing to specified or best-performance default modules.
+
+    """
+    reference = core.get_option('SCF', 'REFERENCE')
+    mtd_type = core.get_global_option('MP_TYPE')  # TODO or ADC_TYPE or CC_TYPE or ...
+    module = core.get_global_option('QC_MODULE')
+    # Considering only adcc/adc
+
+    # TODO Actually one should do selection on a couple of other options here
+    #      as well, e.g. adcc supports frozen-core and frozen-virtual,
+    #      spin-specific states or spin-flip methods.
+
+    func = None
+    if reference == 'RHF':
+        if mtd_type == 'CONV':
+            if module == 'ADCC' and extras.addons("adcc"):
+                func = run_adcc
+            elif module in ['', 'BUILTIN']:
+                func = run_adc
+        # TODO Does the builtin support any other MP type?
+
+    if reference == 'UHF':
+        if mtd_type == 'CONV':
+            if module == 'ADCC' and extras.addons("adcc"):
+                func = run_adcc
+
+    # Note: ROHF is theoretically available in adcc, but are not fully tested
+    #       ... so will be added later.
+
+    if func is None:
+        raise ManagedMethodError(['select_adc2', name, 'MP_TYPE', mtd_type, reference, module])
+
+    if kwargs.pop('probe', False):
+        return
+    else:
+        return func(name, **kwargs)
```

Comment:
thanks for the select fn. logic looks good. can add a table section https://github.com/psi4/psi4/blob/master/doc/sphinxman/source/notes_c.rst if you like.

Reply:
No problem.